packages/fast-jss-manager-angular/src/design-system.directive.ts

```diff
@@ -0,0 +1,47 @@
+import { Directive, ElementRef, Input, SimpleChanges } from "@angular/core";
+import { eventTypes } from "./utilities/get-event-types";
+import jss, { stylesheetManager } from "./jss";
+
+/**
+ * Directive definition
+ */
+@Directive({
+    selector: "design-system",
+})
+
+/**.
+ * Class definition
+ */
+export class DesignSystemDirective {
+    @Input() private config: any;
+
+    private el: HTMLElement;
+    private children: EventTarget[];
+
+    constructor(el: ElementRef) {
+        this.el = el.nativeElement;
+        this.children = [];
+
+        this.el.addEventListener(eventTypes.getConfig, (e: CustomEvent) => {
+            Object.assign(e.detail, this.config);
+        }, true);
+
+        this.el.addEventListener(eventTypes.registerComponent, (e: CustomEvent) => {
+            this.children.push(e.target);
+        }, true);
+
+        this.el.addEventListener(eventTypes.deregisterComponent, (e: CustomEvent) => {
+            this.children.filter((child: EventTarget) => child !== e.target);
+        }, true);
+    }
+
```

Comment:
these functions should include comments to explain what's happening and why it matters.